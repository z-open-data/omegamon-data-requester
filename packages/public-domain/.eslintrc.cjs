/* eslint-disable */
const eslintBaseConfig = require('devtools-configs/eslintBase.cjs');

module.exports = eslintBaseConfig;
