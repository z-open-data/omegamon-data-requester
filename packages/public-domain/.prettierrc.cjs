const prettierConfig = require('devtools-configs/prettier.cjs');

module.exports = prettierConfig;
