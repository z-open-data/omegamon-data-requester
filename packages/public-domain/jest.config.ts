import type { Config } from 'jest';

// It's okay to use undefined variables in this Jest config file because they are provided by the Jest environment
// force timezone to UTC to allow tests to work regardless of local timezone
// generally used by snapshots, but can affect specific tests
process.env.TZ = 'UTC';

const config: Config = {
  // Inform jest to only transform specific node_module packages.
  transform: {
    '^.+\\.(t|j)sx?$': [
      '@swc/jest',
      {
        sourceMaps: 'inline',
        jsc: {
          parser: {
            syntax: 'typescript',
            tsx: true,
            decorators: false,
            dynamicImport: true,
          },
        },
      },
    ],
  },
  prettierPath: null, // this is required for Jest (below v30) inline snapshots to work
  transformIgnorePatterns: ['node_modules/(?!@ngrx|(?!deck.gl)|ng-dynamic)'],

  modulePaths: ['<rootDir>/src/'],
};

export default config;
