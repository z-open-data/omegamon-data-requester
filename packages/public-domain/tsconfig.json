{
  "include": ["**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules"],
  "compilerOptions": {
    "rootDir": "./",
    "baseUrl": "./src",
    "typeRoots": ["./node_modules/@types"],

    "strict": true,
    "noUncheckedIndexedAccess": true,
    "noImplicitOverride": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,

    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "bundler",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true
  }
}
