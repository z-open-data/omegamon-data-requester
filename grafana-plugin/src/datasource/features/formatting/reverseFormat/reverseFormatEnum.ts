import { ColumnMetadata } from 'public-domain';

import { convertToNumberOrThrow } from './convertToNumberOrThrow';

/**
 * Returns according enum key, if provided value is matched to any enum value.
 *
 * E.g.
 * value = 'UNKNOWN'
 * enums = { '0000': 'UNKNOWN', '0001': 'BAD_GATEWAY' }
 * result = 0
 */
export function reverseFormatEnum(
  value: string | number,
  { enums, type }: ColumnMetadata
): number | string | undefined {
  if (!enums) {
    return undefined;
  }
  for (const key in enums) {
    if (enums[key] === value || enums[key] === Number(value)) {
      if (type === 'number') {
        return convertToNumberOrThrow(key);
      }
      return key;
    }
  }
  return undefined;
}
