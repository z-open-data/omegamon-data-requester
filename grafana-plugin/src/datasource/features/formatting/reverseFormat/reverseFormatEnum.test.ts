import { ColumnMetadata, EnumMap } from 'public-domain';

import { reverseFormatEnum } from './reverseFormatEnum';

describe(reverseFormatEnum.name, () => {
  it('returns undefined, if no enum map', () => {
    expect(reverseFormatEnum('bla', { type: 'number', enums: undefined } as ColumnMetadata)).toEqual(undefined);
  });

  it('returns undefined, if value does not match any value in enum map', () => {
    const enums: EnumMap = {
      bla: 'val1',
      bla2: 'val2',
    };
    expect(reverseFormatEnum('someOtherValue', { enums, type: 'string' } as ColumnMetadata)).toEqual(undefined);
  });

  it('returns matched key as string, if column type is string', () => {
    const key = 'blaKey';
    const enums = { [key]: 'blaVal' } as EnumMap;
    expect(reverseFormatEnum('blaVal', { type: 'string', enums } as ColumnMetadata)).toEqual(expect.any(String));
    expect(reverseFormatEnum('blaVal', { type: 'string', enums } as ColumnMetadata)).toEqual(key);
  });

  it('throws error, if column type is number, but matched key cannot be converted to number', () => {
    const val = 'val';
    const enums = { someKeyThatCantBeConvertedToNumber: val } as EnumMap;
    expect(() => reverseFormatEnum(val, { type: 'number', enums } as ColumnMetadata)).toThrowError();
  });

  it('can match value of type string that hold number to a value as simple number in enum map', () => {
    const valueAsNumber = 42;
    const valueAsString = '42';
    const keyAsNumber = 69;
    const enums = { [keyAsNumber]: valueAsNumber } as EnumMap;
    expect(reverseFormatEnum(valueAsString, { type: 'number', enums } as ColumnMetadata)).toEqual(keyAsNumber);
  });

  const enums: EnumMap = {
    '-123': 'val1',
    '0': 'val2',
    '00234': 'val3',
    '0X0000': 'val4',
    '0X0A': 'val5',
    '0xFFFF': 'val6',
  };
  const columnMd = { type: 'number', enums } as ColumnMetadata;

  Object.entries(enums).forEach(([key, val]) => {
    it('returns number, if column type is number and matched key is decimal or hex number', () => {
      expect(reverseFormatEnum(val, columnMd)).toEqual(expect.any(Number));
      expect(reverseFormatEnum(val, columnMd)).toEqual(Number(key));
    });

    it(`raw key = "${key}"`, () => {
      expect(reverseFormatEnum(val, columnMd)).toMatchSnapshot();
    });
  });
});
