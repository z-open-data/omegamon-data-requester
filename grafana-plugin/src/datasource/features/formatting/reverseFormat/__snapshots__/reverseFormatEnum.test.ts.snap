// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reverseFormatEnum raw key = "-123" 1`] = `-123`;

exports[`reverseFormatEnum raw key = "0" 1`] = `0`;

exports[`reverseFormatEnum raw key = "00234" 1`] = `234`;

exports[`reverseFormatEnum raw key = "0X0000" 1`] = `0`;

exports[`reverseFormatEnum raw key = "0X0A" 1`] = `10`;

exports[`reverseFormatEnum raw key = "0xFFFF" 1`] = `65535`;
